go/fn/result.go

```diff
@@ -178,18 +130,69 @@ type Field struct {
 
 type Results []*Result
 
+// ErrorE writes an Error level `result` to the results slice. It accepts arguments according to a format specifier.
+// e.g.
+// results.Errorf("bad kind %v", "invalid")
+func (r *Results) Errorf(format string, a ...any) {
+	errResult := &Result{Severity: Error, Message: fmt.Sprintf(format, a...)}
+	*r = append(*r, errResult)
+}
+
+// ErrorE writes the `error` as an Error level `result` to the results slice.
+// e.g.
+//
+//	err := error.New("test)
+//	results.ErrorE(err)
+func (r *Results) ErrorE(err error) {
+	errResult := &Result{Severity: Error, Message: err.Error()}
+	*r = append(*r, errResult)
+}
+
+// Infof writes an Info level `result` to the results slice. It accepts arguments according to a format specifier.
+// e.g.
+//
+//	results.Infof("update %v %q ", "ConfigMap", "kptfile.kpt.dev")
+func (r *Results) Infof(format string, a ...any) {
+	infoResult := &Result{Severity: Info, Message: fmt.Sprintf(format, a...)}
+	*r = append(*r, infoResult)
+}
+
+// Warningf writes a Warning level `result` to the results slice. It accepts arguments according to a format specifier.
+// e.g.
+//
+//	results.Warningf("bad kind %q", "invalid")
+func (r *Results) Warningf(format string, a ...any) {
+	warnResult := &Result{Severity: Warning, Message: fmt.Sprintf(format, a...)}
+	*r = append(*r, warnResult)
+}
+
+// WarningE writes an error as a Warning level `result` to the results slice.
+// Normally this function can be used for cases that need error tolerance.
+func (r *Results) WarningE(err error) {
```

Comment:
Why did you choose to give these methods this `E` suffix? In my past projects, I've normally seen the `E` in methods that return an error, but that isn't the case here.

Reply:
Yeah, E suffix is normally a convention for wrapping/rebuilding a new error.   
do you have any naming suggestions? 
I was trying to use `Error(err error)` and `Warning(err error)`, but the `Error` function is preserved to implement the  "error.Error" interface (required by function
`[ApplyFnBySelector](https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/78748f08d9977e08b5cf2dd678d04ed22d209a75/go/fn/resourcelist.go#L300`)

Reply:
> Yeah, E suffix is normally a convention for wrapping/rebuilding a new error.
do you have any naming suggestions?

I think `WarningE` is fine, I was just curious your thought process. I would have also suggested `Error` and `Warning`, but your explanation makes sense.